ruint-macro/src/lib.rs

```diff
@@ -120,19 +147,19 @@ fn parse(value: &str, bits: &str) -> Result<(usize, Vec<u64>), String> {
     // Check value range
     if limbs.len() > num_limbs || limbs.last().copied().unwrap_or(0) > mask {
         let value = value.trim_end_matches('_');
-        return Err(format!("Value too large for Uint<{bits}>: {value}"));
+        return Err(format!("Value too large for {base_type}<{bits}>: {value}"));
     }
 
     Ok((bits, limbs))
 }
 
 /// Transforms a [`Literal`] and returns the substitute [`TokenTree`]
-fn transform_literal(literal: Literal) -> TokenTree {
+fn transform_literal(literal: Literal, base_type: LiteralBaseType) -> TokenTree {
     let source = literal.to_string();
-    if let Some((value, bits)) = source.split_once('U') {
-        let tokens = parse(value, bits).map_or_else(
+    if let Some((value, bits)) = source.split_once(base_type.delimiter()) {
```

Comment:
We should read the `base_type` from the literal, so you can freely mix `U256` and `B256` (and soon `I256`) in the contents.